scio-cats/src/main/scala/com/spotify/scio/cats/syntax/SCollectionSyntax.scala

```diff
@@ -0,0 +1,391 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.cats.syntax
+
+import com.spotify.scio.util.Functions.{Fns, ReduceFn}
+import com.spotify.scio.values._
+import com.spotify.scio.coders._
+import cats.Applicative
+import cats.Traverse
+import cats.Functor
+import cats.FunctorFilter
+import cats.FlatMap
+import cats.Foldable
+import cats.Monoid
+import cats.data.Nested
+import cats.kernel.Order
+import org.apache.beam.sdk.transforms.Combine
+
+import java.lang.{Iterable => JIterable}
+import java.util.{List => JList}
+
+import scala.language.higherKinds
+
+// scalastyle:off method.name
+final class CatsOps[A](private val coll: SCollection[A]) extends AnyVal {
+
+  /**
+   * Fold implemented using the given Monoid[A] instance.
+   */
+  def fold_(implicit M: Monoid[A], C: Coder[A]): SCollection[A] =
+    coll.context.wrap {
+      val combine = Combine.globally(new ReduceFn[A] {
+        override val vacoder = Coder[JList[A]]
+        override val vocoder = C
+
+        override def reduceOption(accumulator: JIterable[A]): Option[A] =
+          Fns
+            .reduceOption(accumulator) { (a, b) =>
+              M.combine(a, b)
+            }
+            .orElse(Some(M.empty))
+
+        override def defaultValue(): A = M.empty
+      })
+
+      coll.applyInternal(combine)
+    }
+
+  /**
+   * Return the min as defined by the implicit `Order[T]`.
+   */
+  def min_(implicit O: Order[A], C: Coder[A]): SCollection[A] =
+    coll.context.wrap {
+      val combine = Combine.globally(new ReduceFn[A] {
+        override val vacoder = Coder[JList[A]]
+        override val vocoder = C
+
+        override def reduceOption(accumulator: JIterable[A]): Option[A] =
+          Fns.reduceOption(accumulator)(O.min)
+      })
+
+      coll.applyInternal(combine)
+    }
+
+  /**
+   * Return the max as defined by the implicit `Order[T]`.
+   */
+  def max_(implicit O: Order[A], C: Coder[A]): SCollection[A] =
+    coll.context.wrap {
+      val combine = Combine.globally(new ReduceFn[A] {
+        override val vacoder = Coder[JList[A]]
+        override val vocoder = C
+
+        override def reduceOption(accumulator: JIterable[A]): Option[A] =
+          Fns.reduceOption(accumulator)(O.max)
+      })
+
+      coll.applyInternal(combine)
+    }
+
+}
+
+final class HigherKindCatsOps[F[_], A](private val coll: SCollection[F[A]]) extends AnyVal {
+
+  /**
+   * Replaces the `A` value in `F[A]` with the supplied value.
+   *
+   * Example:
+   *
+   * {{{
+   * scala> val coll: SCollection[List[Int]] = sc.paralelize(Seq(List(1, 2, 3)))
+   * scala> coll.as("hello") // List(hello, hello, hello)
```

Comment:
Let's write this as : `val out: SCollection[List[Int]] = coll.as("hello") // SCollection(List(hello, hello, hello))`

Reply:
I am not sure why I got confused with this.